### ISSUE TITLE
How to draw a polygon with a hole ?

### ISSUE THREAD

**Issue by finscn** [pixijs/pixi.js#3827]
How to draw a polygon with a hole with Graphics?

**Comment by bigtimebuddy** [comment]
Yes, there is an `addHole` method which takes the current shape and sets it as a hole to the previous shape. 

https://jsfiddle.net/ew6nyavx/

This only seems to work when using `moveTo` and then using line, bezier, quadratic line methods on Graphics class. The convenience drawing methods (`drawRect`, `drawCircle`, etc) don't seem to be working as holes.

https://jsfiddle.net/ew6nyavx/2/

**Comment by bigtimebuddy** [comment]
Did some more digging on this issue, seems the holes can only be made if the hole and the source shape are both Polygons. 

**This works**
```js
new PIXI.Graphics()
  .beginFill(0x00ffcc)
  .drawPolygon([0, 0, 200, 0, 200, 300, 0, 300])
  .drawPolygon([10, 10, 190, 10, 190, 290, 10, 290])
  .addHole();
```
**This doesn't work**
```js
new PIXI.Graphics()
  .beginFill(0x00ffcc)
  .drawRect(0, 0, 200, 300)
  .drawRect(10, 10, 180, 280)
  .addHole();
```

**Comment by markknol** [comment]
In case anyone needs a hole in his circle, I used this https://gist.github.com/markknol/5c5d48655ebac555a6eec41792acdfb6
